yt_dlp/extractor/servus.py

```diff
@@ -17,128 +13,73 @@ class ServusIE(InfoExtractor):
+        }, {
+            'url': 'https://www.pm-wissen.com/umwelt/v/aa-28wxenx9s2111/',
+            'only_matching': True,
+        }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url).upper()
 
-        token = self._download_json(
-            'https://auth.redbullmediahouse.com/token', video_id,
-            'Downloading token', data=urlencode_postdata({
-                'grant_type': 'client_credentials',
-            }), headers={
-                'Authorization': 'Basic SVgtMjJYNEhBNFdEM1cxMTpEdDRVSkFLd2ZOMG5IMjB1NGFBWTBmUFpDNlpoQ1EzNA==',
-            })
-        access_token = token['access_token']
-        token_type = token.get('token_type', 'Bearer')
-
+        # We need to specify a timezone here in order for the endpoint to work.
         video = self._download_json(
-            'https://sparkle-api.liiift.io/api/v1/stv/channels/international/assets/%s' % video_id,
-            video_id, 'Downloading video JSON', headers={
-                'Authorization': '%s %s' % (token_type, access_token),
-            })
-
-        formats = []
-        thumbnail = None
-        for resource in video['resources']:
-            if not isinstance(resource, dict):
-                continue
-            format_url = url_or_none(resource.get('url'))
-            if not format_url:
-                continue
-            extension = resource.get('extension')
-            type_ = resource.get('type')
-            if extension == 'jpg' or type_ == 'reference_keyframe':
-                thumbnail = format_url
-                continue
-            ext = determine_ext(format_url)
-            if type_ == 'dash' or ext == 'mpd':
-                formats.extend(self._extract_mpd_formats(
-                    format_url, video_id, mpd_id='dash', fatal=False))
-            elif type_ == 'hls' or ext == 'm3u8':
-                formats.extend(self._extract_m3u8_formats(
-                    format_url, video_id, 'mp4', entry_protocol='m3u8_native',
-                    m3u8_id='hls', fatal=False))
-            elif extension == 'mp4' or ext == 'mp4':
-                formats.append({
-                    'url': format_url,
-                    'format_id': type_,
-                    'width': int_or_none(resource.get('width')),
-                    'height': int_or_none(resource.get('height')),
-                })
-        self._sort_formats(formats)
+            'https://api-player.redbull.com/stv/servus-tv?videoId=%s&timeZone=%s' % (video_id, 'Europe%2FVienna'),
+            video_id,
+            'Downloading video JSON'
+        )
 
-        attrs = {}
-        for attribute in video['attributes']:
-            if not isinstance(attribute, dict):
-                continue
-            key = attribute.get('fieldKey')
-            value = attribute.get('fieldValue')
-            if not key or not value:
-                continue
-            attrs[key] = value
+        formats = self._extract_m3u8_formats(
```

Comment:
Please use self._extract_m3u8_formats_and_subtitles here.
https://www.servustv.com/natur/v/aansszcx3yi9jmlmhdc1/ is one example of a link with subtitles.